src/layout/IHyprLayout.cpp

```diff
@@ -178,6 +185,60 @@ void IHyprLayout::onWindowCreatedFloating(PHLWINDOW pWindow) {
     }
 }
 
+bool IHyprLayout::onWindowCreatedAutoGroup(PHLWINDOW pWindow) {
+    static auto AUTOGROUP = CConfigValue<Hyprlang::INT>("group:auto_group");
+    if ((*AUTOGROUP || g_pInputManager->m_bWasDraggingWindow) // check if auto_group is enabled, or, if the user is manually dragging the window into the group.
+        && g_pCompositor->m_pLastWindow.lock()                // check if a focused window exists.
+        && g_pCompositor->m_pLastWindow != pWindow            // fixes a freeze when activating togglefloat to transform a floating group into a tiled group.
+        && g_pCompositor->m_pLastWindow->m_pWorkspace ==
+            pWindow
+                ->m_pWorkspace // fix for multimonitor: when there is a focused group in monitor 1 and monitor 2 is empty, this enables adding the first window of monitor 2 when using the mouse to focus it.
+        && g_pCompositor->m_pLastWindow->m_sGroupData.pNextWindow.lock()  // check if the focused window is a group
+        && pWindow->canBeGroupedInto(g_pCompositor->m_pLastWindow.lock()) // check if the new window can be grouped into the focused group
+        && !g_pXWaylandManager->shouldBeFloated(pWindow)) {               // don't group XWayland windows that should be floated.
+
+        switch (pWindow->m_bIsFloating) {                           // checks in what mode is the new window being created: float (case true) or tile (case false).
+            case (false):                                           // this would be the case if allfloat=false on the workspace, or a window rule is making the new window to tile.
```

Comment:
these 2 comments are also redundant, the comments for the if condition are ok but this code is clear as day